src/plugins/virtuals.js

```diff
@@ -35,27 +35,69 @@ module.exports = function (Bookshelf) {
       }
     },
 
-    // Passing `{virtuals: true}` or `{virtuals: false}` in the `options`
-    // controls including virtuals on function-level and overrides the
-    // model-level setting
-    toJSON: function(options) {
+  /**
+   * @method toJSON
+   * @description
+   *
+   * Passing `{virtuals: true}` or `{virtuals: false}` in the `options`
+   * controls including virtuals on function-level and overrides the
+   * model-level setting
+   *
+   * Params must be an object like `{virtualName1 : paramsForVirtualName1,
+   * virtualName2 : paramsForVirtualName2}.
+   * paramsForVirtualName can be a value or an array, depending on the virtual
+   * function parameters
+   *
+   *     var PersonWithTitle = bookshelf.Model.extend({
+   *      virtuals: {
+   *        fullName: function(param) {
+   *          return this.get('firstName') + ' ' + param;
+   *        },
+   *        fullNameWithTitle: function(title,surname) {
+   *          return title + ' ' + this.get('firstName') + ' ' + surname;
+   *        }
+   *      }
+   *     })
+   *     var john = new Person({firstName:"John"});
+   *     var json = john.toJSON({fullName:'Wood' , fullNameWithTitle:['Mr.' , 'Wood']};
+   *     //Output : {firstName:'John', fullName:'John Wood', fullNameWithTitle:'Mr. John Wood'}
+   *     console.log(json);
+   *     var json = john.toJSON({virtuals:true},{fullName:'Wood' , fullNameWithTitle:['Mr.' , 'Wood']};
+   *     //Output : {firstName:'John', fullName:'John Wood', fullNameWithTitle:'Mr. John Wood'}
+   *     console.log(json);
+   *     var json = john.toJSON({virtuals:false},{fullName:'Wood' , fullNameWithTitle:['Mr.' , 'Wood']};
+   *     //Output : {firstName:'John'}
+   *     console.log(json);
```

Comment:
Maybe a little bit of spacing between these examples to make things easier to see